version: "3.7"
services:

  nginx:
    image: nginx:1.19.5
    networks:
      - aws_network
    volumes:
      - aws_media:/data/media/
      - aws_static:/data/static/
      - /home/ubuntu/nginx.conf:/etc/nginx/nginx.conf
    ports:
      - 80:80

  django:
    image: aws_django:5
    networks:
      - aws_network
    secrets:
      - DJANGO_SECRET_KEY
      - MARIADB_USER
      - MARIADB_PASSWORD
    volumes:
      - aws_media:/home/GIS_2ban_1/media/
      - aws_static:/home/GIS_2ban_1/staticfiles/

  mariadb:
    image: mariadb:10.5
    networks:
      - aws_network
    volumes:
      - aws_mariadb:/var/lib/mysql
    secrets:
      - MARIADB_USER
      - MARIADB_ROOT_PASSWORD
      - MARIADB_PASSWORD
    environment:
      MARIADB_DATABASE: django
      MARIADB_USER_FILE:  /run/secrets/MARIADB_USER
      MARIADB_PASSWORD_FILE: /run/secrets/MARIADB_PASSWORD
      MARIADB_ROOT_PASSWORD_FILE:  /run/secrets/MARIADB_ROOT_PASSWORD


networks:
  aws_network:

volumes:
  aws_media:
  aws_static:
  aws_mariadb:

secrets:
  DJANGO_SECRET_KEY:
    external: true
  MARIADB_USER:
    external: true
  MARIADB_PASSWORD:
    external: true
  MARIADB_ROOT_PASSWORD:
    external: true
